sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts

```diff
@@ -54,4 +57,56 @@ export class ApplicationService extends RecordDataModelService<Application> {
 
     return updateResult.affected;
   }
+
+  /**
+   * Finds all applications with some pending student assessment to be
+   * processed by the workflow, ignoring applications that already have
+   * some workflows in progress or were already queued for execution.
+   * All pending student assessments will be returned ordered by its creation
+   * date to allow the select of the next one to be executed (usually only one
+   * record would be expected).
+   * @returns applications with pending assessments to be executed.
+   */
+  async getApplicationsToStartAssessments(): Promise<Application[]> {
```

Comment:
just to refer to my memory, if a student has multiple applications waiting for an assessment, was our plan to process the application in parallel?

Reply:
Different applications will be processed in parallel, and assessments from the same application will be processed in order.